frontend/packages/topology/src/components/CreateConnectorWidget.tsx

```diff
@@ -0,0 +1,206 @@
+import * as React from 'react';
+import { observer } from 'mobx-react';
+import { DragEvent, DragSourceMonitor, DragSourceSpec } from '../behavior/dnd-types';
+import { useDndDrag } from '../behavior/useDndDrag';
+import { AnchorEnd, Graph, GraphElement, isGraph, isNode, Node } from '../types';
+import Point from '../geom/Point';
+import Layer from './layers/Layer';
+import { hullPath } from '../utils';
+import { ContextMenu, ContextMenuItem } from './contextmenu';
+
+import './CreateConnectorWidget.scss';
+
+export const CREATE_CONNECTOR_OPERATION = 'createconnector';
+export const CREATE_CONNECTOR_DROP_TYPE = '#createConnector#';
+
+const DEFAULT_HANDLE_ANGLE = 12 * (Math.PI / 180);
+const DEFAULT_HANDLE_LENGTH = 32;
+
+export type ConnectorChoice = {
+  label: string;
+};
+
+export type CreateConnectorOptions = {
+  handleAngle?: number;
+  handleLength?: number;
+};
+
+export type CreateConnectorRendererProps = {
+  startPoint: Point;
+  endPoint: Point;
+  dragging: boolean;
+  hints: string[] | undefined;
+  tipContents?: React.ReactNode;
+  className?: string;
+};
+
+export type CreateConnectorRenderer = React.ComponentType<CreateConnectorRendererProps>;
```

Comment:
This isn't correct. The only required props that are expected of a component used in `withCreateConnector` are:
```
  startPoint: Point,
  endPoint: Point,
  hints: string[] | undefined,
```